xml_converter/src/attribute/color.cpp

```diff
@@ -15,10 +18,17 @@ using namespace std;
 //
 // Parses a Color from the value of a rapidxml::xml_attribute.
 // TODO(#98): Color should be saved in a better format then the raw hex string.
+// TODO: Every node that has a Hex needs an alpha but not every alpha needs Hex
 ////////////////////////////////////////////////////////////////////////////////
 Color parse_color(rapidxml::xml_attribute<>* input, vector<XMLError*>*) {
     Color color;
-    color.hex = get_attribute_value(input);
+    std::string hex = get_attribute_value(input);
+    if (hex.length() == 6 || 7) {
+        if (hex.compare(0, 1, "#")) {
+            hex = hex.erase(0, 1);
+        }
+        color.hex = hex;
+    }
```

Comment:
There should probably be an else here where it adds an error... given that if length != 6 or 7 it is an error. The current code would just silently return the empty Color object deleting the "hex" value.